diff --git a/sound/soc/qcom/qdsp6/audioreach.c b/sound/soc/qcom/qdsp6/audioreach.c
index 241c3b4..c2d0c37 100644
--- a/sound/soc/qcom/qdsp6/audioreach.c
+++ b/sound/soc/qcom/qdsp6/audioreach.c
@@ -202,6 +202,12 @@ struct apm_display_port_module_intf_cfg {
 } __packed;
 #define APM_DP_INTF_CFG_PSIZE ALIGN(sizeof(struct apm_display_port_module_intf_cfg), 8)
 
+struct apm_usb_module_intf_cfg {
+	struct apm_module_param_data param_data;
+	struct usb_audio_intf_cfg cfg;
+} __packed;
+#define APM_USB_INTF_CFG_PSIZE ALIGN(sizeof(struct apm_usb_module_intf_cfg), 8)
+
 struct apm_module_sp_vi_op_mode_cfg {
 	struct apm_module_param_data param_data;
 	struct param_id_sp_vi_op_mode_cfg cfg;
@@ -687,6 +693,59 @@ static int audioreach_display_port_set_media_format(struct q6apm_graph *graph,
 	return q6apm_send_cmd_sync(graph->apm, pkt, 0);
 }
 
+static int audioreach_usb_set_media_format(struct q6apm_graph *graph,
+					   const struct audioreach_module *module,
+					   const struct audioreach_module_config *cfg)
+{
+	struct apm_usb_module_intf_cfg *intf_cfg;
+	struct apm_module_frame_size_factor_cfg *fs_cfg;
+	struct apm_module_param_data *param_data;
+	struct apm_module_hw_ep_mf_cfg *hw_cfg;
+	int ic_sz = APM_USB_INTF_CFG_PSIZE;
+	int ep_sz = APM_HW_EP_CFG_PSIZE;
+	int fs_sz = APM_FS_CFG_PSIZE;
+	int size = ic_sz + ep_sz + fs_sz;
+	void *p;
+
+	struct gpr_pkt *pkt __free(kfree) = audioreach_alloc_apm_cmd_pkt(size, APM_CMD_SET_CFG, 0);
+	if (IS_ERR(pkt))
+		return PTR_ERR(pkt);
+
+	p = (void *)pkt + GPR_HDR_SIZE + APM_CMD_HDR_SIZE;
+
+	hw_cfg = p;
+	param_data = &hw_cfg->param_data;
+	param_data->module_instance_id = module->instance_id;
+	param_data->error_code = 0;
+	param_data->param_id = PARAM_ID_HW_EP_MF_CFG;
+	param_data->param_size = ep_sz - APM_MODULE_PARAM_DATA_SIZE;
+	hw_cfg->mf.sample_rate = cfg->sample_rate;
+	hw_cfg->mf.bit_width = cfg->bit_width;
+	hw_cfg->mf.num_channels = cfg->num_channels;
+	hw_cfg->mf.data_format = module->data_format;
+	p += ep_sz;
+
+	fs_cfg = p;
+	param_data = &fs_cfg->param_data;
+	param_data->module_instance_id = module->instance_id;
+	param_data->error_code = 0;
+	param_data->param_id = PARAM_ID_HW_EP_FRAME_SIZE_FACTOR;
+	param_data->param_size = fs_sz - APM_MODULE_PARAM_DATA_SIZE;
+	fs_cfg->frame_size_factor = 1;
+	p += fs_sz;
+
+	intf_cfg = p;
+	param_data = &intf_cfg->param_data;
+	param_data->module_instance_id = module->instance_id;
+	param_data->error_code = 0;
+	param_data->param_id = PARAM_ID_USB_AUDIO_INTF_CFG;
+	param_data->param_size = ic_sz - APM_MODULE_PARAM_DATA_SIZE;
+	intf_cfg->cfg.usb_token = cfg->usb_token;
+	intf_cfg->cfg.svc_interval = cfg->usb_svc_interval;
+
+	return q6apm_send_cmd_sync(graph->apm, pkt, 0);
+}
+
 /* LPASS Codec DMA port Module Media Format Setup */
 static int audioreach_codec_dma_set_media_format(struct q6apm_graph *graph,
 						 const struct audioreach_module *module,
@@ -1334,6 +1393,10 @@ int audioreach_set_media_format(struct q6apm_graph *graph,
 	case MODULE_ID_DISPLAY_PORT_SINK:
 		rc = audioreach_display_port_set_media_format(graph, module, cfg);
 		break;
+	case MODULE_ID_USB_AUDIO_SINK:
+	case MODULE_ID_USB_AUDIO_SOURCE:
+		rc = audioreach_usb_set_media_format(graph, module, cfg);
+		break;
 	case  MODULE_ID_SMECNS_V2:
 		rc = audioreach_set_module_config(graph, module, cfg);
 		break;
diff --git a/sound/soc/qcom/qdsp6/audioreach.h b/sound/soc/qcom/qdsp6/audioreach.h
index 89f172a..3514017 100644
--- a/sound/soc/qcom/qdsp6/audioreach.h
+++ b/sound/soc/qcom/qdsp6/audioreach.h
@@ -34,6 +34,8 @@ struct q6apm_graph;
 #define MODULE_ID_SPEAKER_PROTECTION	0x070010E2
 #define MODULE_ID_SPEAKER_PROTECTION_VI	0x070010E3
 #define MODULE_ID_OPUS_DEC		0x07001174
+#define MODULE_ID_USB_AUDIO_SINK	0x0700104F
+#define MODULE_ID_USB_AUDIO_SOURCE	0x07001050
 
 #define APM_CMD_GET_SPF_STATE		0x01001021
 #define APM_CMD_RSP_GET_SPF_STATE	0x02001007
@@ -148,6 +150,7 @@ struct payload_pcm_output_format_cfg {
 } __packed;
 
 #define PARAM_ID_ENC_BITRATE			0x08001052
+#define PARAM_ID_USB_AUDIO_INTF_CFG		0x080010D6
 
 struct param_id_enc_bitrate_param {
 	uint32_t bitrate;
@@ -504,6 +507,11 @@ struct param_id_display_port_intf_cfg {
 	uint32_t dptx_idx;
 } __packed;
 
+struct usb_audio_intf_cfg {
+	uint32_t usb_token;
+	uint32_t svc_interval;
+} __packed;
+
 #define PARAM_ID_HW_EP_MF_CFG			0x08001017
 struct param_id_hw_ep_mf {
 	uint32_t sample_rate;
@@ -814,6 +822,8 @@ struct audioreach_module_config {
 	u16	dp_idx;
 	u32	channel_allocation;
 	u32	sd_line_mask;
+	u32	usb_token;
+	u32	usb_svc_interval;
 	int	fmt;
 	struct snd_codec codec;
 	u8 channel_map[AR_PCM_MAX_NUM_CHANNEL];
